src/Datadog.Trace/Configuration/ConfigurationKeys.cs

```diff
@@ -84,6 +84,36 @@ public static class ConfigurationKeys
         /// <seealso cref="TracerSettings.LogsInjectionEnabled"/>
         public const string LogsInjectionEnabled = "DD_LOGS_INJECTION";
 
+        /// <summary>
+        /// Configuration key for setting the number of traces allowed
+        /// to be submitted per second.
+        /// </summary>
+        /// <seealso cref="TracerSettings.MaxTracesSubmittedPerSecond"/>
+        public const string MaxTracesSubmittedPerSecond = "DD_MAX_TRACES_PER_SECOND";
+
+        /// <summary>
+        /// Configuration key for setting custom sampling rules based on regular expressions.
+        /// Semi-colon separated list of sampling rules.
+        /// The rule is matched in order of specification. The first match in a list is used.
+        ///
+        /// Per entry:
+        ///   The item 'rate' is required in decimal format.
+        ///   The item 'service' is optional in regular expression format, to match on service name.
+        ///   The item 'operation' is optional in regular expression format, to match on operation name.
+        ///
+        /// Example:
+        ///   'rate=0.5, service=cart.* ; rate=0.2, operation=web ; rate=1.0, service=background, operation=insert ; rate=0.1'
+        ///
+        /// The example above will match every trace in services that start with the name cart and give them a 50% sample rate.
```

Comment:
Regex note: the example `service=cart.*` matches services that contain `cart` _anywhere_ in the name (e.g. `my-cartoon`). To make it match service names that _start_ with `cart`, users can use something like `^cart`.

Similarly, `operation=web` will match any service with `web` anywhere in the name (e.g. `mywebapp`). To get an exact match, users will need to use `operation=^web$`.

Reply:
I like the idea of adding them automatically. It seems like a much better experience.